### PR TITLE
ci: Rewrite custom code checks in Python and add a license header check

### PR DESCRIPTION
**Pull request by iljakuklic**
* Hopefully more robust and portable
* <s>Will fix the missing license headers once I see it fail in CI to check it works</s> DONE

### CODE REVIEW

**Review by TheQuantumPhysicist — APPROVED** [review]
Thank you for moving from bash to python!

**Review by muursh — APPROVED** [review]

File: .github/scripts/codecheck.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+def run_checks():
+    ok = True
+    ok = ok and disallow(SCALECODEC_RE, exclude = ['serialization/core'])
+    ok = ok and disallow(JSONRPSEE_RE, exclude = ['rpc'])
+    ok = ok and check_licenses()
+    return ok
+
+if __name__ == '__main__':
+    if not run_checks():
+        sys.exit(1)
```
Comment (TheQuantumPhysicist) [inline .github/scripts/codecheck.py:81]: perhaps it's better to raise exceptions on errors. That will show the error on spot and exit the program.

**Review by stanislav-tkach — APPROVED** [review]

File: .github/scripts/codecheck.py (hunk 1 above)
Comment (iljakuklic) [reply .github/scripts/codecheck.py:81]: I wanted to let the program list all code problems it finds during the run, without stopping at first one.

Comment (TheQuantumPhysicist) [reply .github/scripts/codecheck.py:81]: Alrighty. That's fine.

**Pull request merged**